storage_drivers/solidfire_san.go

```diff
@@ -191,18 +191,45 @@ func (d *SolidfireSANStorageDriver) Validate() error {
 	return nil
 }
 
+// Strip storage prefix name to simple Docker Name
+func (d *SolidfireSANStorageDriver) StripVolumePrefix(name string) string {
+	dockerName := name
+	prefix := string(d.Config.CommonStorageDriverConfig.StoragePrefixRaw)
+	if prefix != "" {
+		dockerName = strings.Replace(name, prefix, "", -1)
+	}
+	return dockerName
+}
+
+// Strip snapshot prefix name to simple Docker Name
+func (d *SolidfireSANStorageDriver) StripSnapshotPrefix(name string) string {
+	dockerName := name
+	prefix := string(d.Config.CommonStorageDriverConfig.SnapshotPrefixRaw)
+	if prefix != "" {
+		dockerName = strings.Replace(name, prefix, "", -1)
+	}
+	return dockerName
+}
+
+// Make SolidFire name
+func MakeSolidFireName(name string) string {
+	return strings.Replace(name, "_", "-", -1)
+}
+
 // Create a SolidFire volume
 func (d *SolidfireSANStorageDriver) Create(name string, opts map[string]string) error {
 	log.Debugf("SolidfireSANStorageDriver#Create(%v)", name)
 
 	var req sfapi.CreateVolumeRequest
 	var qos sfapi.QoS
 	var vsz int64
-	var meta = map[string]string{"platform": "Docker-NDVP"}
+	var dName = d.StripVolumePrefix(name)
```

Comment:
Thinking about this more I'm unclear on what's hacky about it, or why it's much different than the fact that the interface is adding a prefix to begin with?

I'd be all for passing in both and using whichever is appropriate, but frankly didn't want to have to touch every backend file to update with this change.  If that's the direction folks want to go that's cool, but I'd suggest we do that as follow up work, perhaps even a major version bump and at least make the driver "functional" with compose and other situations where the _ is used at this point.